assume/common/forecasts.py

```diff
@@ -212,13 +281,33 @@ def calculate_EOM_price_forecast(self):
 
     def calculate_marginal_cost(self, pp_series: pd.Series):
         """
-        Calculates the marginal cost of a power plant based on the fuel costs and efficiencies of the power plant.
-
-        :param pp_series: Series with power plant data
-        :type pp_series: pd.Series
-        :return: the marginal cost of the power plant
-        :rtype: float
+        Calculates the marginal cost of a power plant based on the fuel costs and efficiencies of the
+        power plant.
+
+        Args:
+        - pp_series (pd.Series): Series containing power plant data.
+
+        Returns:
+        - float: The marginal cost of the power plant.
+
+        This method calculates the marginal cost of a power plant by taking into account the following
+        factors:
+        - Fuel costs based on the fuel type and efficiency of the power plant.
+        - Emissions costs considering the CO2 price and emission factor of the power plant.
+        - Fixed costs, if specified for the power plant.
+
+        The calculation involves the following steps:
+        1. Determines the fuel price based on the fuel type of the power plant.
+        2. Calculates the fuel cost by dividing the fuel price by the efficiency of the power plant.
+        3. Calculates the emissions cost based on the CO2 price and emission factor, adjusted by the
+        efficiency of the power plant.
+        4. Considers any fixed costs specified for the power plant.
+        5. Aggregates the fuel cost, emissions cost, and fixed cost to obtain the marginal cost of the
+        power plant.
+
+        Returns the resulting marginal cost as a float value.
```

Comment:
If we want, we can change the docstring format from the default sphinx one:
https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#an-example-class-with-docstrings

to Google Doc strings or Numpy doc strings:
https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google
or 
https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy

which adds the napoleon plugin.
But I think we should use the same code style all over assume.

Reply:
please explain, I don't understand

Reply:
Actually this isn't even google Style docs but something weird inbetween - Probably just a Copilot suggestion?

Reply:
ok, I get it now. So what I have written, it is basically google form with some extra explanation at the end. We can discuss this next time if we wanna keep it and convert existing stuff to it. I find it personally extremely helpful. Let's keep it for now and talk about it during a development meeting.